app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  before_action :logged_in_user, only: [:create]
+
+  def create
+    post = Post.find(params[:comment][:post_id])
+    if params[:comment][:comment_id].present?
+      parent = Comment.find(params[:comment][:comment_id])
+    else
+      parent = post
+    end
+    comment = parent.comments.build
+    if (comment.update_attributes(comment_params))
+      comment.save
```

Comment:
ah, one more thing here, if you call `update_attributes` on an object, there is no need to call `save` on that object (update_attributes saves the record)